File: labs/lab_09/ortiz_damien/p03/p03.cc
// file: /data/courses/ece_1111/2025_01_fall/labs/lab_09/ortiz_damien/p03/p03.cc/
//

// Version History: 
// 20251022 create initial file

// include files 
//
#include "p03.h"

// Program will sort text file using heap sorting
//

// main program
//
int main(int argc, char** argv){


  // check if there are proper number of arguments
  //
  if (argc == 3){

    char* fileName = argv[2];

    char* op = argv[1];

    // find number of lines in file and tell user if file doesn't exist
    //
    int lineCount = countLines(fileName);
    if(lineCount == -1)
      return 1;

    // create array of lines in file
    //
    char** lines = getLineText(fileName);  

    // sort lines using bubble sorting
    //
    mysort(lines, lineCount, op);          

    if (!strcmp(op, "-a") || !strcmp(op, "-d"))
      printString(lines, stdout); 


    // delete allocated memory
    //
    cleanStringArrayMemory(lines);

    // exit gracefully
    //
    return 0;
  }

  // error case for not enough arguments
  //
  fprintf(stderr, "Incorrect usage.\n");
  fprintf(stdout, "Usage: p01.exe [option] <text_file>\n");

  // exit with errors
  //
  return 1;
}

// end of file

File: homework/hw_12/ortiz_damien/p03/p03.py
#!/usr/bin/env python

# file: /data/courses/ece_1111/2025_01_fall/homework/hw_12/ortiz_damien/p03/p03/p03.py
#

# Version History: 
# 20251114 create initial file

# Imports:
#
import sys
import os
import p03_00

# main program
#
def main(argv):

        # check for correct number of args
        #
        if len(argv)==2:
                # create instance of class
                #
                wc = p03_00.WordCounter(argv[1])
                
                # compute the word frequencies
                #
                wc.compute_word_frequencies()
                
                # print our result
                #
                wc.print_frequencies(ascending=True)
        
                # exit gracefully
                #
                return True
        else:
                # give user error messages for incorrect number of arguments
                #
                print("Incorrect number of arguments.")
                print("Usage: p03.py <file>")

                # exit with errors
                #
                return False
                
# begin gracefully
#
if __name__ == '__main__':
	main(sys.argv[0:])
        
# end of file
